drake/solvers/Constraint.h

```diff
@@ -116,12 +116,12 @@ class LinearConstraint : public Constraint {
   virtual void eval(const Eigen::Ref<const Eigen::VectorXd>& x,
                     Eigen::VectorXd& y) const override {
     y.resize(getNumConstraints());
-    y = getMatrix() * x;
+    y = getMatrix() * x * -1;
   }
   virtual void eval(const Eigen::Ref<const TaylorVecXd>& x,
                     TaylorVecXd& y) const override {
     y.resize(getNumConstraints());
-    y = getMatrix().cast<TaylorVarXd>() * x;
+    y = getMatrix().cast<TaylorVarXd>() * x * -1;
```

Comment:
I cannot explain it directly.  However, it does not affect any other solvers, as SnoptSolver never invokes LinearConstraint::eval().

The sign flipping was necessary to make LinearConstraint::eval() work in the same way as the eval() function on the various generic constraints defined in testOptimizationProblem